qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -631,6 +631,16 @@ def on_icon_changed(self, tab, icon):
         if config.val.tabs.tabs_are_windows:
             self.window().setWindowIcon(icon)
 
+    @pyqtSlot(usertypes.KeyMode)
+    def on_mode_entered(self, mode):
+        """Save input mode when tabs.mode_on_change = restore."""
+        input_modes = [usertypes.KeyMode.insert, usertypes.KeyMode.passthrough]
+        if (mode in input_modes and
+                config.val.tabs.mode_on_change == 'restore'):
+            tab = self.currentWidget()
+            if tab and tab is not None:
```

Comment:
Why check for both? `if tab is not None:` should be enough.

Reply:
Well, maybe I should just do "if tab:" since this page says it *can* be 0 as well:

http://doc.qt.io/qt-5/qtabwidget.html#currentWidget

Reply:
That's sure an interesting documentation :laughing: 

> The tab dialog does its best to make sure that this value is never 0 (but if you try hard enough, it can be).

However, it's a pointer - and a nullpointer in C++ means `None` in Python.